R/model-persistence.R

```diff
@@ -243,6 +260,14 @@ unserialize_model <- function(model, custom_objects = NULL, compile = TRUE) {
 export_savedmodel.keras.engine.training.Model <- function(object, export_dir_base, ...) {
   if (!is_backend("tensorflow"))
     stop("'export_savedmodel' is only supported in the TensorFlow backend.")
+
+  if (!is_tensorflow_implementation()) {
+    original_learning_phase <- k_learning_phase()
+    on.exit(function() k_set_learning_phase(original_learning_phase), add = TRUE)
```

Comment:
I don't think the resetting of the learning phase will actually execute (since `on.exit` takes an expression not a function)

Reply:
So, looks like `k_learning_phase()` is a tensor, not the actual value. I tried retrieving this value using something like:

```
tf <- keras::backend()
tf$get_session()$run(original_learning_phase)
```

but the tensor is no initialized an fails as:

```
InvalidArgumentError: You must feed a value for placeholder tensor 'dropout_1/keras_learning_phase' with dtype bool
```

Not sure if it's that feasible to get the actual value, we could track over `k_set_learning_phase()` but someone could still use other mechanisms to update this value.

I think we should leave the behavior as is for now.